### PR TITLE
WFLY-5422 SSO is not destroyed after session timeout period of <distributable/> app

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.jboss.org/browse/WFLY-5422

This PR modifies our distributable session manager to delegate expiration events to undertow session listeners (including SingleSignOnAuthenticationMechanism's SessionInvalidationListener) rather than notifying the application listeners directly.

Includes the test case contained in https://github.com/wildfly/wildfly/pull/8210

### CODE REVIEW

**Comment by bstansberry** [comment]
@pferraro can someone on your team review this? This is a very large change late in the release cycle so someone with more familiarity with the code in question should look at it.

I've started 5 tests each of this branch on master-ignore Linux and master-ignore Windows.

**Comment by pferraro** [comment]
@rhusar @rachmatowicz Please review.
@stuartwdouglas Can you review this as well? since this change involves improving compliance with Undertow's SessionManager contract.

**Comment by bstansberry** [comment]
The 10 custom test runs came back clean.

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/sso/ClusteredSingleSignOnTestCase.java [hunk 1]
```diff
@@ -134,4 +135,18 @@ public void testNoAuthSingleSignOn(
         SSOTestBase.executeNoAuthSingleSignOnTest(new URL(baseURL1, "/"), new URL(baseURL2, "/"), log);
     }
 
+    /**
+     * Test single sign-on is destroyed after session timeout
+     *
+     * Testing https://issues.jboss.org/browse/WFLY-5422
+     */
+    @Test
+    @Ignore("https://issues.jboss.org/browse/WFLY-5422")
```
Comment (rhusar) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/sso/ClusteredSingleSignOnTestCase.java]: ^ should be un-ignored :-)

Comment (pferraro) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/sso/ClusteredSingleSignOnTestCase.java]: Whoops.  This is the test that validates this fix.  Removing @Ignore...

**Comment by pferraro** [comment]
This fix isn't complete - closing for now.

**Pull request closed without merging**